### ISSUE TITLE
Registry: how is the `packageName.js` file created?

### ISSUE THREAD

**Issue by unional** [jspm/jspm-cli#1276]
Hi,

I'm working on a `jspm-p4` registry endpoint for perforce.  I'm able to copy the files to the right folder. I try to learn how to create the `packageName.js` file under the registry root that points to the main file? I looked at all 3 existing endpoint but couldn't figure out how it is done.

````
+-- myendp
    +-- mypackage@0.1.2
          +-- files....
    +-- mypackage@0.1.2.js  <--- this is what I want to create
````

Thanks,

**Comment by guybedford** [comment]
It's created to match the `main` config property returned by the endpoint package config.

**Comment by unional** [comment]
Thanks!  Is it possible to expose the package as typescript? It works now if I set `"main": "index"` or `"main": "index.js"`, but my original intent is to expose the package as typescript (thus my main is `index.ts`).

My plan is to expose all of my internal module (in perforce) in typescript, let systemjs load and transpile it in runtime, and build the final bundle at the application level.  Can I do that or should I adjust my plan?

**Comment by bab5139** [comment]
+1 for the question regarding an all typescript package.  I need support for this if it doesn't already exist.

**Comment by guybedford** [comment]
Non-js mains are currently only supported for plugins. There is an issue at https://github.com/jspm/jspm-cli/issues/1048 tracking this, which will likely be resolved in jspm 0.17.
